src/main/java/hng_java_boilerplate/config/WebSecurityConfig.java

```diff
@@ -71,7 +72,7 @@ public SecurityFilterChain httpSecurity(HttpSecurity httpSecurity) throws Except
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(httpRequests ->
                         httpRequests
-//                                .requestMatchers("").hasAuthority(String.valueOf(Role.ADMIN))
+                                .requestMatchers("/api/v1/products/{id}").hasAuthority(String.valueOf(Role.ROLE_ADMIN))
```

Comment:
you can achieve this by enabling method security and annotating the controller method with secured or preAuthorize.